src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -231,12 +231,14 @@ public void delete(final String eventTypeName, final String timelineId, final Cl
     }
 
     public void deleteAllTimelinesForEventType(final String eventTypeName)
-            throws TopicDeletionException, TimelineException, NotFoundException {
+            throws TopicDeletionException, TopicRepositoryException, TimelineException, NotFoundException {
         LOG.info("Deleting all timelines for event type {}", eventTypeName);
         final List<Timeline> timelines = getTimelines(eventTypeName);
         for (final Timeline timeline : timelines) {
             final TopicRepository topicRepository = getTopicRepository(timeline);
-            topicRepository.deleteTopic(timeline.getTopic());
+            if (topicRepository.topicExists(timeline.getTopic())) {
+                topicRepository.deleteTopic(timeline.getTopic());
```

Comment:
What if between the calls `topicRepository.topicExists` and `topicRepository.deleteTopic` topic will be deleted by the job that removes expired topics? Then you deletion process will fail with exception.
That's why I think it's better to call `deleteTopic` without checking that the topic exists and just catch the exception in case topic deletion failed.